### PR TITLE
CLOUDP-219248: Support --tier in Atlas Stream Instance Create command

### PR DESCRIPTION
**Pull request by andym03**
## Proposed changes

- Introduce a new optional flag, `--tier` for the Atlas Stream Processing create instance command.
- This value defaults to `SP30`, in the coming months we will support `SP10` as well.
  - Note, the public API already supports receiving this value. 

```
To get started quickly, specify a name, a cloud provider, and a region to configure an Atlas Stream Processing instance.To use this command, you must authenticate with a user account or an API key with the Project Owner role.

Usage:
  atlas streams instances create <name> [flags]

Examples:
  # Deploy an Atlas Stream Processing instance called myProcessor for the project with the ID 5e2211c17a3e5a48f5497de3:
  atlas streams instance create myProcessor --projectId 5e2211c17a3e5a48f5497de3 --provider AWS --region VIRGINIA_USA --tier SP30

Flags:
  -h, --help               help for create
  -o, --output string      Output format. Valid values are json, json-path, go-template, or go-template-file. To see the full output, use the -o json option.
      --projectId string   Hexadecimal string that identifies the project to use. This option overrides the settings in the configuration file or environment variable.
      --provider string    Cloud service provider that applies to the provisioned Atlas Stream Processing instance. (default "AWS")
  -r, --region string      Human-readable label that identifies the physical location of your Atlas Stream Processing instance. The region can affect network latency and performance if it is far from your source or sink.
      --tier string        Tier for your Stream Instance. (default "SP30")

Global Flags:
  -P, --profile string   Name of the profile to use from your configuration file. To learn about profiles for the Atlas CLI, see https://dochub.mongodb.org/core/atlas-cli-save-connection-settings.
```

_Jira ticket:_ CLOUDP-219248

## Checklist

- [✅] I have signed the [MongoDB CLA](https://www.mongodb.com/legal/contributor-agreement)
- [✅] I have added tests that prove my fix is effective or that my feature works
- [✅] I have added any necessary documentation in document requirements section listed in [CONTRIBUTING.md](https://github.com/mongodb/mongodb-atlas-cli/blob/master/CONTRIBUTING.md) (if appropriate)
- [N/A] I have addressed the @mongodb/docs-cloud-team comments (if appropriate)
- [N/A] I have updated [test/README.md](https://github.com/mongodb/mongodb-atlas-cli/blob/master/test/README.md) (if an e2e test has been added)
- [✅] I have run `make fmt` and formatted my code

### CODE REVIEW

**Review by jarjee — COMMENTED** [review]
I think we also need to tweak the e2e test for this as well, to make sure that we can pass in the --tier to atlas as well (which should work today).

[The test is this one](https://github.com/mongodb/mongodb-atlas-cli/blob/a429ab6c870a3084fb6408aaee4c81a9fe3001d8/test/e2e/atlas/streams_test.go#L72)

File: internal/cli/streams/instance/create.go
Comment (pulkitkalra-mdb) [inline internal/cli/streams/instance/create.go:51]: if tier isn't passed in - we want to default to SP30 right?

File: internal/cli/streams/instance/create_test.go
Comment (pulkitkalra-mdb) [inline internal/cli/streams/instance/create_test.go:48]: should expected have SP30 as tier here?

File: internal/cli/streams/instance/create.go
Comment (andym03) [reply internal/cli/streams/instance/create.go:51]: That's a good question. The endpoint this calls defaults to SP30. I wonder which is cleaner, the client picking a default or the server.

File: internal/cli/streams/instance/create_test.go
Comment (andym03) [reply internal/cli/streams/instance/create_test.go:48]: Pending discussion on the other comment, the backend will fill out this value. So this test should expect the field not to be completed. 

If we decide the atlas cli should be the one defaulting the value, we can update both tests to expect the field.

File: internal/cli/streams/instance/create.go
Comment (pulkitkalra-mdb) [reply internal/cli/streams/instance/create.go:51]: Oh yeah the server will need the default regardless, so I reckon this is good and we shouldn't need to add it here as well.

File: internal/cli/streams/instance/create_test.go
Comment (andym03) [reply internal/cli/streams/instance/create_test.go]: Both tests now expect tier to be fulfilled; done here! [df8d5c2](https://github.com/mongodb/mongodb-atlas-cli/pull/2839/commits/df8d5c23b9f4736fe11de5ae054df1b209a6c150)

File: internal/cli/streams/instance/create.go
Comment (andym03) [reply internal/cli/streams/instance/create.go:54]: Had a chat with Nathan, ended up adding the default behaviour: [df8d5c2](https://github.com/mongodb/mongodb-atlas-cli/pull/2839/commits/df8d5c23b9f4736fe11de5ae054df1b209a6c150)

Comment (andym03) [reply internal/cli/streams/instance/create.go:54]: Reason for having both; users could call the atlas api without using the cli.

**Review by jarjee — APPROVED** [review]
LGTM, please wait for an APIX engineer to also LGTM before merging

File: internal/cli/streams/instance/create.go
Comment (gssbzn) [reply internal/cli/streams/instance/create.go:54]: this will never be empty since you are setting it with 
```go
cmd.Flags().StringVar(&opts.tier, flag.Tier, "SP30", usage.StreamsInstanceTier)
```

File: internal/cli/streams/instance/create_test.go
Comment (gssbzn) [inline internal/cli/streams/instance/create_test.go:121]: update the original test should test all available flags

Comment (andym03) [reply internal/cli/streams/instance/create_test.go:121]: Done in [292f8ad](https://github.com/mongodb/mongodb-atlas-cli/pull/2839/commits/292f8adf0500b4d189ad960b11c59c80ec717794) !

Comment (gssbzn) [reply internal/cli/streams/instance/create_test.go:121]: not seeing the change on that commit

Comment (andym03) [reply internal/cli/streams/instance/create_test.go]: So sorry about this, total mind blank. I've removed `TestCreateBuilderWithTier` and instead just have `TestCreateBuilder` with all arguments. Fixed the wrong test in a rush 😅 

[5df54a2](https://github.com/mongodb/mongodb-atlas-cli/pull/2839/commits/5df54a234371d95b95f992615eb123574a679697)

**Review by gssbzn — APPROVED** [review]
LGTM

**Pull request merged**
